File: 0060-permutation-sequence/0060-permutation-sequence.java
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        factorial = [1] * n
        for i in range(1, n):
            factorial[i] = factorial[i-1] * i
    
        k -= 1
        nums = list(range(1, n + 1))
        permutation = []

        for i in range(n, 0, -1):
            index = k // factorial[i - 1]
            k %= factorial[i - 1]
            permutation.append(nums[index])
            nums.pop(index)
    
        return ''.join(map(str, permutation))

File: 0060-permutation-sequence/0060-permutation-sequence.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        numbers = list(range(1, n + 1))
        # Adjust k to be zero-indexed
        k -= 1
        permutation = []

        # Generate permutation
        for i in range(n, 0, -1):
            # Determine the index of the current position
            factorial = math.factorial(i - 1)
            index = k // factorial
            k %= factorial

            # Append the number and remove it from the list
            permutation.append(numbers[index])
            numbers.pop(index)

        return ''.join(map(str, permutation))